File: lib/javascript/test/fixtures/journey-summaries.ts
import type { JourneySummary } from "../../src/journey-summary";

export const journeySummaries: { encoded: string; decoded: JourneySummary }[] =
	[
		{
			encoded: "00002ADC091B010200D2020000630054",
			decoded: {
				previous: undefined,
				lastPaidAt: { year: 2021, month: 6, day: 28, hour: 9, minute: 27 },
				consecutivePayments: 1,
				cardType: "AvanzaTopUp",
				free: false,
				route: 210,
				direction: 2,
				transfersLeft: 0x63,
			},
		},
		{
			encoded: "D2012AD90D33010200D20200006300AE",
			decoded: {
				previous: { route: 210, direction: 1 },
				lastPaidAt: { year: 2021, month: 6, day: 25, hour: 13, minute: 51 },
				consecutivePayments: 1,
				cardType: "AvanzaTopUp",
				free: false,
				route: 210,
				direction: 2,
				transfersLeft: 0x63,
			},
		},
		{
			encoded: "23023454102D0102001601000063000B",
			decoded: {
				previous: { route: 35, direction: 2 },
				lastPaidAt: { year: 2026, month: 2, day: 20, hour: 16, minute: 45 },
				consecutivePayments: 1,
				cardType: "AvanzaTopUp",
				free: false,
				route: 22,
				direction: 1,
				transfersLeft: 0x63,
			},
		},
		{
			encoded: "16013468002B04020016020000630011",
			decoded: {
				previous: { route: 22, direction: 1 },
				lastPaidAt: { year: 2026, month: 3, day: 8, hour: 0, minute: 43 },
				consecutivePayments: 4,
				cardType: "AvanzaTopUp",
				free: false,
				route: 22,
				direction: 2,
				transfersLeft: 0x63,
			},
		},
		{
			encoded: "1F013518160B010D01D2020000620091",
			decoded: {
				previous: { route: 31, direction: 1 },
				lastPaidAt: { year: 2026, month: 8, day: 24, hour: 22, minute: 11 },
				consecutivePayments: 1,
				cardType: "LazoTopUp",
				free: true,
				route: 210,
				direction: 2,
				transfersLeft: 0x62,
			},
		},
	];

File: lib/rust/tests/spec/fixtures/journey_summaries.rs
use zgz_transport::{CardType, Date, Direction, JourneySummary, LastPaidAt, Leg, Route};

fn paid(year: u16, month: u8, day: u8, hour: u8, minute: u8) -> LastPaidAt {
	LastPaidAt {
		date: Date { year, month, day },
		hour,
		minute,
	}
}

fn leg(route: u8, direction: Direction) -> Leg {
	Leg {
		route: Route(route),
		direction,
	}
}

pub fn journey_summaries() -> Vec<(&'static str, JourneySummary)> {
	let summary = |previous,
	               last_paid_at,
	               consecutive_payments,
	               card_type,
	               free,
	               route,
	               direction,
	               transfers_left| {
		JourneySummary {
			previous,
			last_paid_at,
			consecutive_payments,
			card_type,
			free,
			route: Route(route),
			direction,
			transfers_left,
		}
	};
	let avanza = CardType::AvanzaTopUp;
	vec![
		(
			"00002ADC091B010200D2020000630054",
			summary(
				None,
				paid(2021, 6, 28, 9, 27),
				1,
				avanza,
				false,
				210,
				Direction::Two,
				0x63,
			),
		),
		(
			"D2012AD90D33010200D20200006300AE",
			summary(
				Some(leg(210, Direction::One)),
				paid(2021, 6, 25, 13, 51),
				1,
				avanza,
				false,
				210,
				Direction::Two,
				0x63,
			),
		),
		(
			"23023454102D0102001601000063000B",
			summary(
				Some(leg(35, Direction::Two)),
				paid(2026, 2, 20, 16, 45),
				1,
				avanza,
				false,
				22,
				Direction::One,
				0x63,
			),
		),
		(
			"16013468002B04020016020000630011",
			summary(
				Some(leg(22, Direction::One)),
				paid(2026, 3, 8, 0, 43),
				4,
				avanza,
				false,
				22,
				Direction::Two,
				0x63,
			),
		),
		(
			"1F013518160B010D01D2020000620091",
			summary(
				Some(leg(31, Direction::One)),
				paid(2026, 8, 24, 22, 11),
				1,
				CardType::LazoTopUp,
				true,
				210,
				Direction::Two,
				0x62,
			),
		),
	]
}
